login/service.go

```diff
@@ -699,14 +700,33 @@ func (keycloak *KeycloakOAuthProvider) CreateOrUpdateKeycloakUser(accessToken st
 		}
 		// let's update the existing user with the fullname, email and avatar from Keycloak,
 		// in case the user changed them since the last time he/she logged in
-		fillUser(claims, user)
-		err = keycloak.Users.Save(ctx, user)
+		fillUser(claims, user, identity)
```

Comment:
Let's check if User or Identity actaully changed. If so then update them. If not then we can skip it:
fillUser(claims *keycloakTokenClaims, user *account.User, identity *account.Identity) (userChanged, identityChanged bool, error)